File: app/src/main/java/com/example/app/recyclerview/SimpleNumberAdapter.java
package com.example.app.recyclerview;

import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.app.R;

/**
 * @author rainfool
 * @date 2018/1/16
 */
public class SimpleNumberAdapter extends RecyclerView.Adapter<SimpleNumberAdapter.SimpleNumberViewHolder> {

    private static final String TAG = "SimpleNumberAdapter";

    private int count = 5;

    public SimpleNumberAdapter(int count) {
        this.count = count;
    }

    @Override
    public SimpleNumberViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        Log.d(TAG, "onCreateViewHolder: sub");
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View view = inflater.inflate(R.layout.item_recycler_demo_sub, parent, false);
        return new SimpleNumberViewHolder(view);
    }

    @Override
    public void onBindViewHolder(SimpleNumberViewHolder holder, int position) {
        Log.d(TAG, "onBindViewHolder: sub" + position);
        holder.setNum(position);
    }

    @Override
    public int getItemCount() {

        return count;
    }

    @Override
    public int getItemViewType(int position) {
        return 1;
    }


    public class SimpleNumberViewHolder extends RecyclerView.ViewHolder {

        TextView textView;

        public SimpleNumberViewHolder(View itemView) {
            super(itemView);
            textView = (TextView) itemView;
        }

        public void setNum(int i) {
            textView.setText(i + "");
        }
    }
}

File: app/src/main/java/com/example/app/recyclerview/SimpleNumberAdapter.kt
package com.example.app.recyclerview

import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.example.app.R

/**
 * @author rainfool
 * @date 2018/1/16
 */
class SimpleNumberAdapter : RecyclerView.Adapter<SimpleNumberAdapter.SimpleNumberViewHolder> {

    private val TAG = "SimpleNumberAdapter"

    private var parentPosition = 0
    private var count = 5

    constructor(count: Int) {
        this.count = count
    }

    constructor(parentPosition: Int, count: Int) {
        this.parentPosition = parentPosition
        this.count = count
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SimpleNumberViewHolder {
        Log.e(TAG, "onCreateViewHolder: sub[$parentPosition]")
        val inflater = LayoutInflater.from(parent.context)
        val view = inflater.inflate(R.layout.item_recycler_demo_sub, parent, false)
        return SimpleNumberViewHolder(view)
    }

    override fun onBindViewHolder(holder: SimpleNumberViewHolder, position: Int) {
        Log.d(TAG, "onBindViewHolder: sub[$parentPosition,$position]")
        holder.setNum(parentPosition*10 + position)
    }

    override fun getItemCount(): Int {

        return count
    }

    override fun getItemViewType(position: Int): Int {
        return 1
    }


    inner class SimpleNumberViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        private var textView: TextView = itemView.findViewById(R.id.tv_simple_demo)

        fun setNum(i: Int) {
            textView.text = i.toString() + ""
        }
    }
}
